### DISCUSSION TITLE
Database node as bottleneck, scalability

### DISCUSSION THREAD

**Discussion by starnutoditopo** [BretFisher/ama#207] [Q&A] [answered]
Supposing we have a swarm containing several application nodes.
All of them access a common database (let's say a Sql server db), running on another node, for storage.
In such a scenario, it looks like the db MUST be unique, in order to ensure consistency.
As an example, consider the voting system of assignment 66 ( https://www.udemy.com/course/docker-mastery/learn/lecture/6790714#overview ).

For a large application, this configuration may be a bottleneck. Is there a way to make the db scale?

Would it be correct to have several replicas of the db, sharing the same volume for data?

**Comment by BretFisher** [comment, marked as answer]
How you scale a database has almost nothing to do with containers or swarm. DB performance or redundancy needs are specific to the DB technology you’re using. Docker and Swarm don’t change that. SQL systems typically can’t share files.
